src/Raven.Server/Smuggler/Documents/Actions/DatabaseRecordActions.cs

```diff
@@ -650,6 +664,29 @@ public async ValueTask WriteDatabaseRecordAsync(DatabaseRecord databaseRecord, S
             result.DatabaseRecord.SnowflakeEtlsUpdated = true;
         }
 
+        if (databaseRecord.EmbeddingsGenerations.Count > 0 && databaseRecordItemType.HasFlag(DatabaseRecordItemType.EmbeddingsGenerations))
+        {
+            if (_log.IsInfoEnabled)
+                _log.Info("Configuring Embedding Generation tasks configuration from smuggler");
+
+            foreach (var etl in databaseRecord.EmbeddingsGenerations)
+            {
+                _currentDatabaseRecord?.EmbeddingsGenerations.ForEach(x =>
```

Comment:
Is this safe to do?
It _looks_ like it may race in some cases, where the `DeleteOngoingTaskCommand` may run after the `AddEmbeddingsGenerationCommand`

Reply:
[RavenDB-23866](https://issues.hibernatingrhinos.com/issue/RavenDB-23866) Check if during restore or import we do not have a race condition when deleting and adding ongoing tasks